[Unordererd & OrderedList] Design guidelines (#309)
primer/brand: apps/docs/content/components/OrderedList/index.mdx

---
title: Ordered list

> description: Use the ordered list component to render a list of ordered items.

Reviewer: Suggestion around deduping the word ordered...

```suggestion
description: Use the ordered list component to display a list of numbered items.
```

---

import ComponentLayout from '../../../src/layouts/component-layout'
export default ComponentLayout

## Anatomy

> ![An image showing an ordered list with different bullet points.](https://github.com/primer/brand/assets/912236/5cda0c94-c120-4fc6-8306-8b55012dee0a)

Reviewer: Clarification as they aren't bullets..

```suggestion
An image showing several sentences as a linear list of numbered items
```

## Usage

Use an ordered list to represent a list of items in a numerical order. Each item within the ordered list represents an item that is ordered in a sequence and displays a number to indicate its position in the sequence.

The ordered list is commonly used when presenting sequential or step-by-step information. It is suitable for instructions, procedures, tutorials, or any content that follows a specific order. Ensure that the items in the ordered list are logically organized and follow a clear progression. This helps readers understand the sequence of information and makes it easier to follow.

<DoDontContainer>
  <Do>

>     <img src="https://github.com/primer/brand/assets/912236/170e2098-4c9a-41ce-b8de-7b1e1403b3a4" />

Reviewer: missing alt text here?

Author: Good point. The caption bellow should work as descriptive text, I'll add the `role="presentation"` to the `<img>`

    <Caption>Use an ordered list to present a sequence of steps.</Caption>
  </Do>
  <Dont>
    <img src="https://github.com/primer/brand/assets/912236/6b014079-1939-454a-8d04-d4c432c4b3e1" />
    <Caption>
      Don't use an ordered list to present a list of items that don't require a
      specific order. Use an unordered list instead.
    </Caption>
  </Dont>
</DoDontContainer>

> Avoid overusing ordered lists. Reserve them for situations where the order of items is genuinely meaningful or where a clear sequence needs to be conveyed. For

Reviewer: Could this line move above the do/don't image as it's related to that and not the guidance beneath it?

lists that don't require a specific order, consider using an [unordered list](/components/UnorderedList) instead.

The ordered list should have conscise and short items. Avoid using long sentences or paragraphs as they can be difficult to scan and comprehend. If you need to include a long item or even headings and links, consider using a [pillar](/components/Pillar), a [card](/components/Card) or a [river](/components/River) instead.

<DoDontContainer>
  <Do>

>     <img src="https://github.com/primer/brand/assets/912236/16fff4c6-8780-4b00-9e78-f76d9c1d91ea" />

Author: that's weird, I changed them in previous commits to this:

<img width="1159" alt="Screenshot 2023-06-21 at 16 11 01" src="https://github.com/primer/brand/assets/912236/25e28b53-bbf8-4382-9970-b9ff2f34d885">

>     <img src="https://github.com/primer/brand/assets/912236/16fff4c6-8780-4b00-9e78-f76d9c1d91ea" />

Author: The preview looks good 🤔 : https://primer-36144985fb-26139705.drafts.github.io/components/OrderedList#usage

>     <img src="https://github.com/primer/brand/assets/912236/16fff4c6-8780-4b00-9e78-f76d9c1d91ea" />

Reviewer: Ohh sorry, i meant generally for all images.. I can still see the description on the example above it.

also to help save even more space and keep it focussed... do we need the headings there?

![Screenshot 2023-06-21 at 15 18 33](https://github.com/primer/brand/assets/13340707/18e98e3c-b405-4b3b-93ac-f80d8c8c2df0)

Non-blocking btw, more a nice-to-have..

---
title: Ordered list
description: Use the ordered list component to display a list of numbered items.
---

import ComponentLayout from '../../../src/layouts/component-layout'
export default ComponentLayout

## Anatomy

> ![An image showing several sentences as a linear list of numbered items](https://github.com/primer/brand/assets/912236/5cda0c94-c120-4fc6-8306-8b55012dee0a)

Author: That's a good point. I added the alt text based on what I've seen in the design guidelines but it's true it's probably not necesary.

Avoid overusing ordered lists. Reserve them for situations where the order of items is genuinely meaningful or where a clear sequence needs to be conveyed. For
lists that don't require a specific order, consider using an [unordered list](/components/UnorderedList) instead.

<DoDontContainer>
  <Do>
    <img
      src="https://github.com/primer/brand/assets/912236/170e2098-4c9a-41ce-b8de-7b1e1403b3a4"
      alt=""
    />
    <Caption>Use an ordered list to present a sequence of steps.</Caption>
  </Do>
  <Dont>
    <img
      src="https://github.com/primer/brand/assets/912236/6b014079-1939-454a-8d04-d4c432c4b3e1"
      alt=""
    />
    <Caption>
      Don't use an ordered list to present a list of items that don't require a
      specific order. Use an unordered list instead.
    </Caption>
  </Dont>
</DoDontContainer>

> The ordered list should have conscise and short items. Avoid using long sentences or paragraphs as they can be difficult to scan and comprehend. If you need to include a long item or even headings and links, consider using a [pillar](/components/Pillar), a [card](/components/Card) or a [river](/components/River) instead.

Reviewer: ```suggestion
The ordered list should have concise and short items. Avoid using long sentences or paragraphs as they can be difficult to scan and comprehend. If you need to include a long item or even headings and links, consider using a [pillar](/components/Pillar), a [card](/components/Card) or a [river](/components/River) instead.
```

<DoDontContainer>
  <Do>
    <img
      src="https://github.com/primer/brand/assets/912236/181f1cb9-05bc-4010-a80b-a978b7ffbe56"
      alt=""
    />
    <Caption>Use an ordered list for short and concise content items.</Caption>
  </Do>
  <Dont>
    <img
      src="https://github.com/primer/brand/assets/912236/28950f6a-d7f0-4d9d-bb70-02c062d309f3"
      alt=""
    />
    <Caption>
      Don't use an ordered list for long and rich sentences, paragraphs,
      headings or links. Use a pillar, card or river components instead.
    </Caption>
  </Dont>
</DoDontContainer>

> ## Options

Reviewer 2: This is empty, do you want to add something or actually just remove the heading?
